### PR TITLE
Global error handling

### PR DESCRIPTION
**Pull request by bdeak4**
This PR adds global express error handling.

Fixes issue #14

### CODE REVIEW

**Review by nkelava — CHANGES_REQUESTED** [review]
In my opinion it would be better if you can add global HTTP status codes and error messages so in case of any change we only have one place to modify and the code will be more consistent.

File: backend/src/consts/httpStatusCodes.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+const httpStatusCodes = {
```
Comment (abasic) [inline backend/src/consts/httpStatusCodes.ts]: There are [npm packages ](https://npms.io/search?q=http+status)that can be used for this.

File: backend/src/app.ts [hunk 2]
```diff
@@ -29,6 +31,16 @@ app.use(
   ),
 );
 
+// Express does not pick up this method if next is not last parameter even if it is unused
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+app.use((err: Error | { error: Error; status: number }, req: Request, res: Response, next: NextFunction) => {
+  if (err instanceof Error) {
+    res.status(httpStatusCodes.INTERNAL_SERVER_ERROR).json({ error: err.message });
+    return;
```
Comment (abasic) [inline backend/src/app.ts]: Is there a specific reason why this return is all alone and not a part of the previous line?

**Review by abasic — CHANGES_REQUESTED** [review]
A couple of minor things. Other than that it looks good!

**Review by nkelava — APPROVED** [review]
Looks good now 👍

**Review by abasic — APPROVED** [review]

**Pull request merged**